src/analyzer/protocol/ldap/ldap.spicy

```diff
@@ -980,16 +1066,69 @@ type AbandonRequest = unit(inout message: Message) {
 #-----------------------------------------------------------------------------
 # Extended Operation
 # https://tools.ietf.org/html/rfc4511#section-4.12
+type ExtendedRequest = unit(inout message: Message, ctx: Ctx&) {
+  var requestValue: bytes;
+  header: ASN1::ASN1Header &requires=($$.tag.class == ASN1::ASN1Class::ContextSpecific);
+  requestName: bytes &size=self.header.len.len &convert=$$.decode(spicy::Charset::ASCII) {
+    message.obj = $$;
+  }
 
-# TODO: implement ExtendedRequest
-# type ExtendedRequest = unit(inout message: Message) {
-#
-# };
+  # If there's more byte to parse, it's the requestValue.
+  switch ( message.opLen > self.offset() ) {
+    True -> : ASN1::ASN1Message(False) &requires=($$.head.tag.class == ASN1::ASN1Class::ContextSpecific) {
+      self.requestValue = $$.application_data;
+    }
+    * -> : void;
+  };
 
-# TODO: implement ExtendedResponse
-# type ExtendedResponse = unit(inout message: Message) {
-#
-# };
+
+  on %done {
+   # Did the client request StartTLS?
+   #
+   # https://datatracker.ietf.org/doc/html/rfc4511#section-4.14.1
+   if ( self.requestName == "1.3.6.1.4.1.1466.20037" )
+     ctx.startTlsRequested = True;
+  }
+};
+
+#-----------------------------------------------------------------------------
+type ExtendedResponseEntry = unit(inout r: ExtendedResponse) {
+  : ASN1::ASN1Message(False) &requires=($$.head.tag.class == ASN1::ASN1Class::ContextSpecific) {
+    if ( $$.head.tag.type_ == ASN1::ASN1Type(10) )
+      r.responseName = $$.application_data;
+    else if ( $$.head.tag.type_ == ASN1::ASN1Type(11) )
+      r.responseValue = $$.application_data;
+    else
+      throw "Unhandled extended response tag %s" % $$.head.tag;
+  }
+};
+
+#-----------------------------------------------------------------------------
+type ExtendedResponse = unit(inout message: Message, ctx: Ctx&) {
+  var responseName: bytes;
+  var responseValue: bytes;
+  : Result {
+    message.result_ = $$;
+  }
+
+  # Try to parse two ASN1 entries if there are bytes left. Both
+  # are optional and identified by context specific tagging.
+  switch ( message.opLen > self.offset() ) {
+    True -> : ExtendedResponseEntry(self);
+    * -> : void;
+  };
```

Comment:
I'd write this with a field `if`, e.g.,

```suggestion
    : ExtendedResponseEntry(self) if (message.opLen > self.offset());
```

Similar for the `ExtendedResponseEntry` below.